engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -167,8 +187,18 @@ public void delete(final long key, final JobRecord record) {
 
   @Override
   public void fail(final long key, final JobRecord updatedValue) {
-    final State newState = updatedValue.getRetries() > 0 ? State.ACTIVATABLE : State.FAILED;
-    updateJob(key, updatedValue, newState);
+    if (updatedValue.getRetries() > 0) {
+      if (updatedValue.getRetryBackOff() > 0) {
+        jobTypeKey.wrapBuffer(updatedValue.getTypeBuffer());
```

Comment:
:x: We need to wrap the job key instead of the job type.

`backoffJobKey` is a composite key of `backoffKey` and `jobKey`. We need to wrap `jobKey` before putting a new entry in `backoffColumnFamily`.

My bad. I made a wrong suggestion in the last review :see_no_evil: 

```suggestion
        jobKey.wrapLong(key);
```